### ISSUE TITLE
Add support for Azure Pipelines

### ISSUE THREAD

**Issue by Abhishek-Dev09** [nexB/scancode-toolkit#1577]
Before starting your migration,we must follow:
a)Create your first pipeline
b)Key concepts for new Azure Pipelines users
c)Building GitHub repositories

Reference: https://docs.microsoft.com/en-us/azure/devops/pipelines/migrate/from-travis?view=azure-devops

**Comment by pombredanne** [comment]
the thing we would need is to:
1. test on Python 2.7 and Python 3.6
2. test on mac, windows and linux.
3. possibly break down the tests in smaller chunks as they are today in the travis and appeyor config, so that the tests are running on multiple parallel instances. That's a nice to have though.

**Comment by pombredanne** [comment]
Basic support has been added and merged with #1581 and 79ae6a11c41c41e15c5bf743f603b7f383051353

**Comment by pombredanne** [comment]
So the Pythons provided by Azure end up being useless https://developercommunity.visualstudio.com/content/problem/598264/known-issue-azure-pipelines-images-missing-sqlite3.html 
Until they fix and roll out new images (a few weeks) the workaround is to install your own python :| 

At this stage I have still not been able to a get a single full test suite run working on Azure on windows, linux or mac :| even though the same test suites (and more) pass on a vanilla windows, ubuntu 16 and macos box and on these 3 OSes on travis and appevyor


Also we have the narrow vs. wide unicode build issue on Linux/Ubuntu (they use a narrow build instead of the default wide build available on Ubuntu).

And the fact that their images install Python in non standard paths.

There are so many quirks there that I wonder if it might not be simpler and easier to use containers for Linux.

And I wonder whether even the Windows and macOS images they have are any good and representative of anything that would be a meaningful setup.

**Comment by pombredanne** [comment]
All tests are now passing on all OS/CI/Pythons combos :relieved:
